app/services/irs_attempts_api/event.rb

```diff
@@ -0,0 +1,65 @@
+module IrsAttemptsApi
+  Event = Struct.new(:jti, :iat, :event_type, :encrypted_event_data, keyword_init: true) do
```

Comment:
I would consider making this a plain ruby object, not a struct, so we could encourage the use of `.build` or `#initialize` instead of competing with the built-in initializer from the struct?

Reply:
Yeah, I actually went back and forth and then said to myself, "I'll bet Zach will say this should be a struct" lol. I can rebuild it as a PoR, I do think that is a little cleaner.

I do like having an init method that lets you set the JTI and IAT b/c that will be needed to construct one from a JSON blob. `.build` is a convenience to not need to create the IAT and JTI in the caller.

Reply:
LOL maybe the Zach in your mind was right

My thinking is that with a Struct, it's easy to create an incomplete/partial object, but if we have a plain object, we can control how it's initialized more, make sure it's not missing any fields, etc etc

Reply:
It may be a moot point. The JWE path I'm working down now makes the `Event` object kinda useless. Replacing that with pairs of JTIs and JWEs. Maybe for the best because the name collision with the `Event` model was also causing problems.